application.go

```diff
@@ -1,35 +1,31 @@
-package client
+package sdk
```

Comment:
Do you think that the `Application` client should be part of the SDK?

IMO, the implementation should live at the consumer level.

Reply:
Yes, I think `ApplicationClient` should be one of the features the SDK provides. 
Reaching out to a node and fetching the list of all applications, handling/wrapping any potential errors, validating the node's response (to any extent that is feasible/reasonable), and even (potentially) declaring types independent of the underlying protocol package, are all expected from the SDK.

Reply:
@adshmh Can you provide this level of context/detail in the PR description next time? (Even just copy-pasting it in this one)

Reply:
Sure, done and will do the same in the future PRs.